[Lens] Migrate by-term `palettes` to `colorMapping` configs

nickofthyme opened the pull request:
## Summary

This PR does the following:

- Switches the `colorMapping` **Tech Preview** toggle switch to a use legacy palettes toggle switch.
- Adds info tooltip to the use legacy palette toggle switch to explain it's future replacement.
- Adds a conversion from legacy palette to new colorMapping config.
- When disabling legacy palette mode, we now generate an equivalent colorMapping config from the current palette.
- The `palette` prop is still preserved to revert back when legacy mode is enabled. This should be removed when palettes are fully deprecated and replaced.

https://github.com/user-attachments/assets/e1edc992-8c6a-442f-b488-fde08fb3b900

> Note: Colors for the 3-stop gradient palettes are close but not exactly equivalent to the legacy palette as the color mapping requires a static color for the middle stop but eui palettes blend the middle color differently based on the number of colors requested.

Closes #221384

### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)


## Release note

The **Color Mapping** feature is now GA. Previous Lens `palette` definitions are deprecated and will continue to function normally with no visual change to existing visualizations. Toggling off legacy mode will replace the palette with an equivalent color mapping configuration.

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

nickofthyme committed:
chore: fix theme access from theme$

nickofthyme committed:
chore: migrate by-term palettes to color mapping config

- remove tech preview toggle for color mapping
- migrate all palettes in tagcloud, xy, datatable and partition to colorMapping config at runtime
- fixup palette definitions and aliases to be used with legacy palette names
- use shared `ColorMappingByTerms` component in tagcloud and partition logic.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299190)
* Commit: 07ee35df29fb6cb913fc2a939fdeef14851df045

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/299190#0196a721-8d7f-4f5f-b9f6-568c00d32fa4)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299190","number":299190,"commit":"07ee35df29fb6cb913fc2a939fdeef14851df045"}],"number":299190}
buildkite-pr-comment-->

nickofthyme committed:
Merge branch 'main' into legacy-palette-color-mapping

elasticmachine commented:
Pinging @elastic/kibana-visualizations (Team:Visualizations)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302893)
* Commit: 05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/302893#0197141c-99c3-446a-a5c1-e24b9f3361a6)

### History
* :broken_heart: [Build #299190](https://buildkite.com/elastic/kibana-pull-request/builds/299190) failed 07ee35df29fb6cb913fc2a939fdeef14851df045

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302893","number":302893,"commit":"05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299190","number":299190,"commit":"07ee35df29fb6cb913fc2a939fdeef14851df045"}],"number":302893}
buildkite-pr-comment-->

nickofthyme committed:
test: fix type errors and add conversion tests

nickofthyme committed:
Merge branch 'main' into legacy-palette-color-mapping

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/303200)
* Commit: 21236b823ac9967f92a3fa8b18a49b2a0d637ee4

### Failed CI Steps
* [FTR Configs #94](https://buildkite.com/elastic/kibana-pull-request/builds/303200#01971871-a458-40cc-a323-21b6845d2f50)
* [FTR Configs #132](https://buildkite.com/elastic/kibana-pull-request/builds/303200#01971871-a48b-4091-9c4f-a79b5341a6a6)
* [FTR Configs #132](https://buildkite.com/elastic/kibana-pull-request/builds/303200#0197187c-a1c6-4ef9-bce2-41bab9b7e2ba)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/303200#01971871-a458-40cc-a323-21b6845d2f50) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/303200/jobs/01971871-a458-40cc-a323-21b6845d2f50/artifacts/0197187f-d10b-4d41-969a-99f328a844bd) FTR Configs #<span></span>94 / dashboard sync colors should sync colors on dashboard for legacy default palette
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/303200#01971871-a48b-4091-9c4f-a79b5341a6a6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/303200/jobs/01971871-a48b-4091-9c4f-a79b5341a6a6/artifacts/0197187c-7c2a-4aa3-aebd-b4992385d84c) FTR Configs #<span></span>132 / lens app - group 4 lens color palette tests should allow to pick legacy color palette in xy chart
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/303200#0197187c-a1c6-4ef9-bce2-41bab9b7e2ba) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/303200/jobs/0197187c-a1c6-4ef9-bce2-41bab9b7e2ba/artifacts/01971884-29e2-487b-9f59-b4cc6a760679) FTR Configs #<span></span>132 / lens app - group 4 lens color palette tests should allow to pick legacy color palette in xy chart

### History
* :broken_heart: [Build #302893](https://buildkite.com/elastic/kibana-pull-request/builds/302893) failed 05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0
* :broken_heart: [Build #299190](https://buildkite.com/elastic/kibana-pull-request/builds/299190) failed 07ee35df29fb6cb913fc2a939fdeef14851df045

cc @nickofthyme

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303200","number":303200,"commit":"21236b823ac9967f92a3fa8b18a49b2a0d637ee4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302893","number":302893,"commit":"05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299190","number":299190,"commit":"07ee35df29fb6cb913fc2a939fdeef14851df045"}],"number":303200}
buildkite-pr-comment-->

nickofthyme committed:
chore: remove palette runtime conversions

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nickofthyme committed:
chore: bring palettes back to dimension editors

- put back palette usage for dim editors
- still unify shared component usage
- revert changes to color mapping UI toggle
- convert palette to color mapping config on toggle

nickofthyme committed:
chore: negate toggle switch to use legacy palettes instead of use colorMapping

nickofthyme committed:
test: update lens page for switch changes

nickofthyme committed:
chore: replace deprecation badge with info tooltip

nickofthyme committed:
Merge branch 'main' into legacy-palette-color-mapping

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/303801)
* Commit: 404ee043639716643cfc1e80460e83f17a14b696

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/303801#0197221d-12cf-4c93-86f7-5c5feeac79dc)

### History
* :broken_heart: [Build #303200](https://buildkite.com/elastic/kibana-pull-request/builds/303200) failed 21236b823ac9967f92a3fa8b18a49b2a0d637ee4
* :broken_heart: [Build #302893](https://buildkite.com/elastic/kibana-pull-request/builds/302893) failed 05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0
* :broken_heart: [Build #299190](https://buildkite.com/elastic/kibana-pull-request/builds/299190) failed 07ee35df29fb6cb913fc2a939fdeef14851df045

cc @nickofthyme

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303801","number":303801,"commit":"404ee043639716643cfc1e80460e83f17a14b696"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303200","number":303200,"commit":"21236b823ac9967f92a3fa8b18a49b2a0d637ee4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302893","number":302893,"commit":"05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299190","number":299190,"commit":"07ee35df29fb6cb913fc2a939fdeef14851df045"}],"number":303801}
buildkite-pr-comment-->

nickofthyme committed:
chore: fix missing props

markov00 committed:
Merge branch 'main' into legacy-palette-color-mapping

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/304440)
* Commit: a21c244e01ff9d2557b3afefe2295be1057c9afd

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-0fde-4156-94b1-2aedd33bd10a)
* [FTR Configs #43](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-0ff9-4475-8d00-1c8389c3ac02)
* [FTR Configs #82](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-1019-4b03-8cce-3096d95965c1)
* [FTR Configs #107](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-1029-4cad-acd3-7bcfe9d99f6b)
* [FTR Configs #118](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-1031-4b0d-81d7-5b68d33f6298)
* [FTR Configs #125](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-1035-4bf9-a8ad-2d0cf328fea4)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/304440#019734ce-0fde-4156-94b1-2aedd33bd10a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/304440/jobs/019734ce-0fde-4156-94b1-2aedd33bd10a/artifacts/019734f2-e389-42fb-a143-52eaf48e7927) Jest Tests #<span></span>11 / AssigneesColumn displays the show less avatars button when the show more is clicked

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `charts` | 160 | 162 | +2 |
| `eventAnnotationListing` | 590 | 592 | +2 |
| `expressionGauge` | 185 | 187 | +2 |
| `expressionHeatmap` | 189 | 191 | +2 |
| `expressionLegacyMetricVis` | 143 | 145 | +2 |
| `expressionMetricVis` | 144 | 146 | +2 |
| `expressionPartitionVis` | 213 | 215 | +2 |
| `expressionTagcloud` | 178 | 180 | +2 |
| `expressionXY` | 262 | 264 | +2 |
| `infra` | 1465 | 1467 | +2 |
| `lens` | 1432 | 1434 | +2 |
| `observability` | 1299 | 1301 | +2 |
| `visDefaultEditor` | 236 | 238 | +2 |
| `visTypeTimeseries` | 407 | 409 | +2 |
| `visualizations` | 601 | 603 | +2 |
| total |  |  | +30 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `@kbn/coloring` | 170 | 172 | +2 |
| `charts` | 254 | 255 | +1 |
| total |  |  | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `charts` | 17.0KB | 4.2KB | -12.8KB |
| `expressionHeatmap` | 15.5KB | 17.1KB | +1.6KB |
| `expressionPartitionVis` | 37.5KB | 37.8KB | +313.0B |
| `expressionTagcloud` | 17.0KB | 17.3KB | +313.0B |
| `expressionXY` | 99.7KB | 100.0KB | +313.0B |
| `lens` | 1.5MB | 1.5MB | -1.7KB |
| `visTypeTimeseries` | 438.5KB | 438.8KB | +313.0B |
| total |  |  | -11.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `charts` | 20.4KB | 36.2KB | +15.7KB |
| `lens` | 59.8KB | 59.8KB | +19.0B |
| `visDefaultEditor` | 29.7KB | 30.0KB | +313.0B |
| total |  |  | +16.1KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `@kbn/coloring` | 224 | 227 | +3 |
| `charts` | 270 | 271 | +1 |
| total |  |  | +4 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `charts` | 4 | 3 | -1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 336 | 337 | +1 |

#### Unreferenced deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/a21c244e01ff9d2557b3afefe2295be1057c9afd) | diff |
| --- | --- | --- | --- |
| `lens` | 0 | 2 | +2 |


</details>


### History
* :broken_heart: [Build #303801](https://buildkite.com/elastic/kibana-pull-request/builds/303801) failed 404ee043639716643cfc1e80460e83f17a14b696
* :broken_heart: [Build #303200](https://buildkite.com/elastic/kibana-pull-request/builds/303200) failed 21236b823ac9967f92a3fa8b18a49b2a0d637ee4
* :broken_heart: [Build #302893](https://buildkite.com/elastic/kibana-pull-request/builds/302893) failed 05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0
* :broken_heart: [Build #299190](https://buildkite.com/elastic/kibana-pull-request/builds/299190) failed 07ee35df29fb6cb913fc2a939fdeef14851df045

cc @nickofthyme

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304440","number":304440,"commit":"a21c244e01ff9d2557b3afefe2295be1057c9afd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303801","number":303801,"commit":"404ee043639716643cfc1e80460e83f17a14b696"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303200","number":303200,"commit":"21236b823ac9967f92a3fa8b18a49b2a0d637ee4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302893","number":302893,"commit":"05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299190","number":299190,"commit":"07ee35df29fb6cb913fc2a939fdeef14851df045"}],"number":304440}
buildkite-pr-comment-->

markov00 committed:
update text

markov00 reviewed (approved):
Just two small nitpicks

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/304579)
* Commit: 24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `charts` | 160 | 162 | +2 |
| `eventAnnotationListing` | 590 | 592 | +2 |
| `expressionGauge` | 185 | 187 | +2 |
| `expressionHeatmap` | 189 | 191 | +2 |
| `expressionLegacyMetricVis` | 143 | 145 | +2 |
| `expressionMetricVis` | 144 | 146 | +2 |
| `expressionPartitionVis` | 213 | 215 | +2 |
| `expressionTagcloud` | 178 | 180 | +2 |
| `expressionXY` | 262 | 264 | +2 |
| `infra` | 1465 | 1467 | +2 |
| `lens` | 1432 | 1434 | +2 |
| `observability` | 1299 | 1301 | +2 |
| `visDefaultEditor` | 236 | 238 | +2 |
| `visTypeTimeseries` | 407 | 409 | +2 |
| `visualizations` | 601 | 603 | +2 |
| total |  |  | +30 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `@kbn/coloring` | 170 | 172 | +2 |
| `charts` | 254 | 255 | +1 |
| total |  |  | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `charts` | 17.0KB | 4.2KB | -12.8KB |
| `dashboard` | 620.6KB | 620.6KB | +7.0B |
| `expressionHeatmap` | 15.5KB | 17.1KB | +1.6KB |
| `expressionPartitionVis` | 37.5KB | 37.8KB | +313.0B |
| `expressionTagcloud` | 17.0KB | 17.3KB | +313.0B |
| `expressionXY` | 99.7KB | 100.0KB | +313.0B |
| `lens` | 1.5MB | 1.5MB | -1.7KB |
| `visTypeTimeseries` | 438.5KB | 438.8KB | +313.0B |
| total |  |  | -11.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `charts` | 20.4KB | 36.2KB | +15.7KB |
| `lens` | 59.8KB | 59.8KB | +19.0B |
| `visDefaultEditor` | 29.7KB | 30.0KB | +313.0B |
| total |  |  | +16.1KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `@kbn/coloring` | 224 | 227 | +3 |
| `charts` | 270 | 271 | +1 |
| total |  |  | +4 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `charts` | 4 | 3 | -1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 336 | 337 | +1 |

#### Unreferenced deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/5a82e1fd95c4e96ee4ca07a1bfcb153a72fe6b5c) | [after](https://github.com/elastic/kibana/commit/24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e) | diff |
| --- | --- | --- | --- |
| `lens` | 0 | 2 | +2 |


</details>


### History
* :yellow_heart: [Build #304440](https://buildkite.com/elastic/kibana-pull-request/builds/304440) was flaky a21c244e01ff9d2557b3afefe2295be1057c9afd
* :broken_heart: [Build #303801](https://buildkite.com/elastic/kibana-pull-request/builds/303801) failed 404ee043639716643cfc1e80460e83f17a14b696
* :broken_heart: [Build #303200](https://buildkite.com/elastic/kibana-pull-request/builds/303200) failed 21236b823ac9967f92a3fa8b18a49b2a0d637ee4
* :broken_heart: [Build #302893](https://buildkite.com/elastic/kibana-pull-request/builds/302893) failed 05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0
* :broken_heart: [Build #299190](https://buildkite.com/elastic/kibana-pull-request/builds/299190) failed 07ee35df29fb6cb913fc2a939fdeef14851df045

cc @nickofthyme

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304579","number":304579,"commit":"24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304440","number":304440,"commit":"a21c244e01ff9d2557b3afefe2295be1057c9afd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303801","number":303801,"commit":"404ee043639716643cfc1e80460e83f17a14b696"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303200","number":303200,"commit":"21236b823ac9967f92a3fa8b18a49b2a0d637ee4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302893","number":302893,"commit":"05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299190","number":299190,"commit":"07ee35df29fb6cb913fc2a939fdeef14851df045"}],"number":304579}
buildkite-pr-comment-->

nreese reviewed (approved):
kibana-presentation changes LGTM
code review only

nickofthyme committed:
chore: remove unused properties

nickofthyme committed:
Merge branch 'main' into legacy-palette-color-mapping

nickofthyme commented on x-pack/platform/plugins/shared/lens/public/runtime_state/converters/raw_color_mappings/utils.ts:
```diff
@@ -9,10 +9,13 @@ import { DatatableColumnType } from '@kbn/expressions-plugin/common';
 export interface ColumnMeta {
   fieldType?: string | 'multi_terms' | 'range';
   dataType?: GenericIndexPatternColumn['dataType'] | DatatableColumnType;
+  isBucketed: boolean;
```
Good catch, this was only used in the `palette` to `colorMapping` runtime converters which were removed in e5e4185e4c116d994c35643a6ac0935a4acde9e6.

Removed `isBucketed` in d363153e61aff41f78fdefa1cacc13b2fc943ae8.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/304794)
* Commit: d73e77eadc06a6f006d95f9425e438ccd6cf93c4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `charts` | 160 | 162 | +2 |
| `eventAnnotationListing` | 590 | 592 | +2 |
| `expressionGauge` | 185 | 187 | +2 |
| `expressionHeatmap` | 189 | 191 | +2 |
| `expressionLegacyMetricVis` | 143 | 145 | +2 |
| `expressionMetricVis` | 144 | 146 | +2 |
| `expressionPartitionVis` | 213 | 215 | +2 |
| `expressionTagcloud` | 178 | 180 | +2 |
| `expressionXY` | 262 | 264 | +2 |
| `infra` | 1465 | 1467 | +2 |
| `lens` | 1435 | 1437 | +2 |
| `observability` | 1299 | 1301 | +2 |
| `visDefaultEditor` | 236 | 238 | +2 |
| `visTypeTimeseries` | 407 | 409 | +2 |
| `visualizations` | 601 | 603 | +2 |
| total |  |  | +30 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `@kbn/coloring` | 170 | 172 | +2 |
| `charts` | 254 | 255 | +1 |
| total |  |  | +3 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `charts` | 17.0KB | 4.2KB | -12.8KB |
| `dashboard` | 620.6KB | 620.6KB | +7.0B |
| `expressionHeatmap` | 15.5KB | 17.1KB | +1.6KB |
| `expressionPartitionVis` | 37.5KB | 37.8KB | +313.0B |
| `expressionTagcloud` | 17.0KB | 17.3KB | +313.0B |
| `expressionXY` | 99.7KB | 100.0KB | +313.0B |
| `lens` | 1.5MB | 1.5MB | -1.8KB |
| `visTypeTimeseries` | 438.5KB | 438.8KB | +313.0B |
| total |  |  | -11.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `charts` | 20.4KB | 36.2KB | +15.7KB |
| `lens` | 59.8KB | 59.8KB | +19.0B |
| `visDefaultEditor` | 29.7KB | 30.0KB | +313.0B |
| total |  |  | +16.1KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `@kbn/coloring` | 224 | 227 | +3 |
| `charts` | 270 | 271 | +1 |
| total |  |  | +4 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `charts` | 4 | 3 | -1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 340 | 341 | +1 |

#### Unreferenced deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/434187fb6ad0d111b9555a24e055667c5079235d) | [after](https://github.com/elastic/kibana/commit/d73e77eadc06a6f006d95f9425e438ccd6cf93c4) | diff |
| --- | --- | --- | --- |
| `lens` | 0 | 2 | +2 |


</details>


### History
* :green_heart: [Build #304579](https://buildkite.com/elastic/kibana-pull-request/builds/304579) succeeded 24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e
* :yellow_heart: [Build #304440](https://buildkite.com/elastic/kibana-pull-request/builds/304440) was flaky a21c244e01ff9d2557b3afefe2295be1057c9afd
* :broken_heart: [Build #303801](https://buildkite.com/elastic/kibana-pull-request/builds/303801) failed 404ee043639716643cfc1e80460e83f17a14b696
* :broken_heart: [Build #303200](https://buildkite.com/elastic/kibana-pull-request/builds/303200) failed 21236b823ac9967f92a3fa8b18a49b2a0d637ee4
* :broken_heart: [Build #302893](https://buildkite.com/elastic/kibana-pull-request/builds/302893) failed 05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0
* :broken_heart: [Build #299190](https://buildkite.com/elastic/kibana-pull-request/builds/299190) failed 07ee35df29fb6cb913fc2a939fdeef14851df045

cc @nickofthyme

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304794","number":304794,"commit":"d73e77eadc06a6f006d95f9425e438ccd6cf93c4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304579","number":304579,"commit":"24b9eb92fdbdacc5b18613ead9b4b0f8d926f16e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304440","number":304440,"commit":"a21c244e01ff9d2557b3afefe2295be1057c9afd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303801","number":303801,"commit":"404ee043639716643cfc1e80460e83f17a14b696"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303200","number":303200,"commit":"21236b823ac9967f92a3fa8b18a49b2a0d637ee4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302893","number":302893,"commit":"05baabfd7223e1e871f608d7bd6d1a9e7b6f75e0"}],"number":304794}
buildkite-pr-comment-->
